doc/howto/edit_your_dockerfile.md

```diff
@@ -0,0 +1,49 @@
+# How to edit your app's Dockerfile
+
+To set up custom programs for your app, such as third-party tools, you can edit your Dockerfile, which contains the setup code needed to build up your container.
+
+A helpful resource is the [Dockerfile Best Practices](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices) guide.
+
+The base KBase Docker image contains a KBase Ubuntu image, but not much else. You will need to include any dependencies, including the installation of whatever tool you are wrapping, to build a custom Docker image that can run your Module.
+
+For example:
+
+```
+RUN \
+  git clone https://github.com/voutcn/megahit.git && \
+  cd megahit && \
+  git checkout tags/v1.0.3 && \
+  make
+```
+
+Note: you do not have to add any lines to the Dockerfile for installing your SDK Module code, as this will happen automatically. The contents of your SDK Module git repo will be added at /kb/module.
+
+It is also important to note that layers in the Docker image are generated by each command in the Dockerfile. To make a streamlined Docker image which will deploy on a worker node more quickly and make cycles of Docker image builds faster, it is best to remove any large, extraneous files, such as a tarball from which you installed a tool.
+
+To accomplish this, commands to your Dockerfile should look like this:
+
+```
+RUN \
+  wget blah.com/tarball.tgz && \
+  tar xzf tarball.tgz && \
+  cd tarball && \
+  make && \
+  cd .. && \
+  rm -rf tarball.tgz
+```
+
+instead of this (**Bad**):
+
+```
+RUN wget blah.com/tarball.tgz
+RUN tar xzf tarball.tgz
+RUN cd tarball
+RUN make
+RUN cd ..
+RUN rm -rf tarball.tgz
+```
+
+because the latter does not remove the previous bloating layers, despite the "rm" command.
```

Comment:
Might be better to call out that each "RUN" command adds an additional layer, which is why so many shell commands are chained together in a single RUN statement. Also point out that the && operator means if any commands returns a exit code that isn't strictly a success code, the chain of commands will stop at that point.